buildchow/todo.adoc

```diff
@@ -0,0 +1,69 @@
+= Advantages noticed within a week:
+
+* fixing JRE across all jobs
+* relasing target platform without breaking builds
+* using 9.0 instead of 4.3.mars for stream
+
+= Local Jenkins Configuration
+
+Things I needed to do to run/test on local Jenkins (not complete yet!)
+
+. Add label to have it be recognized as a runnable node. (i.e. RHEL7)
+. Add jdk jdk1.7 and jdk1.6 to have a server to look up.
+. Add maven installers
+. enable safe html under security
+. Add native_tools / sep (todo)
+. Install the plugins! (looking into using something like https://github.com/maxandersen/jenkins-ootb)
+ 
+= Todos
+
+Open:
+. buildpublish to jboss.org does not require name but publisher jenkins does (? new deploy might make this invalid)
+
+Done:
+. support downstream-ext https://wiki.jenkins-ci.org/display/JENKINS/Downstream-Ext+Plugin (done: https://review.openstack.org/#/c/158117/)
+. be able to turn builder on/off (done: `disabled: true|false`)
+. handle JAVA17 property so its not a variable. (done: not used anymore in original jenkins)
+. do tests timeout (done: just using normal build timeout)
+. check if pattern are correct. (done: using variable to handle difference of the names)
+. missing AbortOperation for timeout (done: Seems abort is the default so not needed to mimic generated config.xml)
+. missing xvnc (done: https://review.openstack.org/#/c/156086/)
+. update via internal server result in cert error (done: https://review.openstack.org/#/c/178794/)
+. add way to not having to add direct support to jjb for a plugin (done: https://review.openstack.org/#/c/160158/)
+
+= Questions
+
+Some of these have been answered - just kept here until verified nothing missed yet.
+
+why no email notificiaton on version watch ? 
+
+why no triggers on version watch ? 
+
+why does versionwatch job define the include/excludes and not in our git repo so versioned/tracable togheter with rest of config ?
+
+why does product ignore failures and errors in mvn build step ?
+
+what does "targetPlatformGroup=jbdevstudio" mean ?
+
+why does product not build its updatesites in mvn instead of separate shell ?
+
```

Comment:
yes, I don't grok why it is doing this:

```
 # move the non-pack200 site zip into the temporary folder
               mv ${{WORKSPACE}}/sources/product/site/target/site_assembly.zip* ${{WORKSPACE}}/tmp-target/
               # purge the target folder built w/o pack200
               rm -fr ${{WORKSPACE}}/sources/product/site/target
               # move the temporary folder back to where it should be
               mv ${{WORKSPACE}}/tmp-target ${{WORKSPACE}}/sources/product/site/target
               
               # replace the update-core zip + MD5 with the new one
               cp -f ${{WORKSPACE}}/sources/product/site/target/site_assembly.zip ${{WORKSPACE}}/sources/product/results/target/jboss-devstudio-*updatesite-core.zip
               z=`find ${{WORKSPACE}}/sources/product/results/target/ -name "jboss-devstudio-*updatesite-core.zip"`
               for m in $(md5sum ${{z}}); do if [[ $m != ${{z}} ]]; then echo $m &gt; ${{z}}.MD5; fi; done
```
what is this doing that tycho build couldn't ? it just seem to be moving the exact same bits/content back and forth,
slightly mutating it and making the result we publish not what maven produces but what jenkins scripts make. which is generally a bad thing.